### PR TITLE
refactor: refactor optimizer parameter

### PR DESCRIPTION
**Pull request by JoanFM**

### CODE REVIEW

File: jina/optimizers/parameters.py [hunk 1]
```diff
@@ -19,6 +23,16 @@ def __init__(
         else:
             self.jaml_variable = jaml_variable
 
+    @abstractmethod
```
Comment (bwanglzu) [inline jina/optimizers/parameters.py:26]: have you set abc as meta class?

Comment (JoanFM) [reply jina/optimizers/parameters.py:26]: no, is it needed?

Comment (bwanglzu) [reply jina/optimizers/parameters.py:26]: yes otherwise it won't work

Comment (JoanFM) [reply jina/optimizers/parameters.py:26]: done!

**Review by bwanglzu — APPROVED** [review]
lgtm!

**Pull request merged**